alacritty/src/display.rs

```diff
@@ -724,3 +761,38 @@ fn compute_cell_size(config: &Config, metrics: &crossfont::Metrics) -> (f32, f32
         ((metrics.line_height + offset_y) as f32).floor().max(1.),
     )
 }
+
+/// Calculate the padding size based on dpr.
+#[inline]
+fn compute_padding_size(config: &Config, dpr: f64) -> (f64, f64) {
+    let padding = config.ui_config.window.padding;
+    (f64::from(padding.x) * dpr, f64::from(padding.y) * dpr)
+}
+
+/// Whether we should set the window's size ourselves or leave it to the windowing system.
+#[inline]
+fn should_set_window_size(config: &Config) -> bool {
+    let dimensions = config.ui_config.window.dimensions;
+    dimensions.columns_u32() != 0
+        && dimensions.lines_u32() != 0
+        && config.ui_config.window.startup_mode != StartupMode::Maximized
+}
+
+/// Get the physical size of a window given padding, terminal dimensions and cell dimensions
+/// infomation.
+fn window_size(
+    dimensions: Dimensions,
+    padding_x: f64,
+    padding_y: f64,
+    cell_width: f32,
+    cell_height: f32,
+) -> PhysicalSize<u32> {
+    // Calculate new size based on cols/lines specified in config.
+    let grid_width = cell_width as u32 * dimensions.columns_u32().max(MIN_COLS as u32);
+    let grid_height = cell_height as u32 * dimensions.lines_u32().max(MIN_LINES as u32);
+
+    let width = padding_x.mul_add(2., f64::from(grid_width)).floor();
+    let height = padding_y.mul_add(2., f64::from(grid_height)).floor();
+
+    PhysicalSize::new(width as u32, height as u32)
+}
```

Comment:
What is the reason for shuffling everything around to these methods?

Reply:
`should_set_window_size` and `window_size` was `GlyphCache::calculate_dimensions`. Now that we don't care whether we should set window size ourselves when we are calculating the minimum window size, I think it makes sense to separate them into 2 functions. I thought I would need to compute the padding size in `window_size`, and if that's the case the logic in `compute_padding_size` would occur at at least 3 places, so it made sense to create a function for it. It turns out that `window_size` doesn't need to calculate the padding size itself, but I thought extracting the padding size logic was not a bad change, as such logic already occurs twice in the original code, so I kept the change.

Reply:
> I thought extracting the padding size logic was not a bad change, as such logic already occurs twice in the original code

I mean if it's not necessary right now, then it's not necessary, right? If it's necessary in the future, we can leave that up to the future.

Reply:
Okay I'll change `compute_padding_size` related code back.

Reply:
I found that `compute_padding_size` actually is called at 3 different places after my change. I guess when some logic occurs 3 times we should give it a function so I'm currently leaving it there.